pkg/ingester/ingester.go

```diff
@@ -333,6 +333,7 @@ type Ingester struct {
 	ingestReader              *ingest.PartitionReader
 	ingestPartitionID         int32
 	ingestPartitionLifecycler *ring.PartitionInstanceLifecycler
+	ingestPartitionWatcher    *ring.PartitionRingWatcher
```

Comment:
I see it only being used in tests to check on the owned partitions. Can we instead move the code for creating a partition watcher (what you removed in a couple of tests) in a utility function and call that? It feels strange to keep test state in production code.

Reply:
The ring watcher is also used by the limiter. I've just stored it on the `Ingester` struct too, but it's already passed ty (and used by) the ingester.

Reply:
i meant the field in the Ingester struct more than the constructor parameter. I don't see `Ingester.ingestPartitionWatcher` being used outside of tests

Reply:
Done in https://github.com/grafana/mimir/pull/7486/commits/3c05bf3bbfb41e2a6e8b02255dc4adf52976eee9.